perceval/backends/core/github.py

```diff
@@ -120,83 +102,174 @@ def fetch(self, from_date=DEFAULT_DATETIME):
+                elif raw_item == '{ASSIGNEE}':
+                    assignee = self.__fetch_assignee_from_cache(cache_items)
+                    current_issue['assignee_data'] = assignee
+                elif raw_item == '{COMMENTS}':
+                    comments = self.__fetch_comments_from_cache(cache_items)
+                    current_issue['comments_data'] = comments
+                elif raw_item == '{}':
+                    yield current_issue
+                    current_issue = next(issues)
+                    self.__init_extra_issue_fields(current_issue)
+
             except StopIteration:
-                if issues:
-                    for issue in self.__build_issues(issues):
-                        yield issue
                 break
 
-            item = json.loads(raw_item)
-
-            if 'login' in item:
-                try:
-                    raw_orgs = next(cache_items)
-                except StopIteration:
-                    # Fatal error. Cache should had stored an organizations item
-                    # per parsed user.
-                    cause = "cache is exhausted but more items were expected"
-                    raise CacheError(cause=cause)
-
-                item['organizations'] = json.loads(raw_orgs)
-                self._users[item['login']] = item
-                continue
-
-            # A new set of issues has been read. It means we already
-            # have the enough information to build and return the
-            # previous set
-            if issues:
-                for issue in self.__build_issues(issues):
-                    yield issue
-
-            # Next issues to parse
-            issues = item
-
-    def __build_issues(self, issues):
-        for issue in issues:
-            for field in ['user', 'assignee']:
-                issue[field + '_data'] = {}
-                if issue[field]:
-                    issue[field + '_data'] = \
-                        self._users[issue[field]['login']]
-            yield issue
+    def __get_issue_comments(self, issue_number):
+        """ Get issue comments """
+
+        comments = []
+        group_comments = self.client.get_issue_comments(issue_number)
+
+        for raw_comments in group_comments:
+            self._push_cache_queue('{COMMENTS}')
+            self._push_cache_queue(raw_comments)
+            self._flush_cache_queue()
+            for comment in json.loads(raw_comments):
+                comment['user_data'] = self.__get_user(comment['user']['login'])
+                comments.append(comment)
+
+        return comments
+
+    def __get_issue_assignee(self, raw_assignee):
+        """ Get issue assignee """
+
+        self._push_cache_queue('{ASSIGNEE}')
+        self._push_cache_queue(raw_assignee)
+        self._flush_cache_queue()
+        assignee = self.__get_user(raw_assignee['login'])
+
+        return assignee
+
+    def __get_user(self, login):
```

Comment:
Has this method change? There is a cache in memory to avoid many trips to the server when requesting for users. You can find the dict in `self._users`. This method should check that dict first.

Reply:
No it hasn't. This method calls
 ```python 
self.client.get_user(login)
```
 which takes care of caching the users

Reply:
Ok!